File: src/com/ociweb/xml/XMLUtilTest.java
package com.ociweb.xml;

import static org.junit.Assert.*;

import org.junit.Test;

/**
 * This class provides unit test methods for XMLUtil.
 * 
 * Copyright 2008 R. Mark Volkmann
 * 
 * This file is part of WAX.
 *
 * WAX is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published
 * by the Free Software Foundation, either version 3 of the License,
 * or (at your option) any later version.
 *
 * Foobar is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty
 * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with WAX.  If not, see http://www.gnu.org/licenses.
 * 
 * @author R. Mark Volkmann, Object Computing, Inc.
 */
public class XMLUtilTest {

    @Test(expected=IllegalArgumentException.class)
    public void testBadComment() {
        XMLUtil.verifyComment("one -- two");
    }
    
    @Test(expected=IllegalArgumentException.class)
    public void testBadNMToken() {
        XMLUtil.verifyNMToken("1a");
    }
    
    @Test(expected=IllegalArgumentException.class)
    public void testBadVersion() {
        XMLUtil.verifyVersion("1.3");
    }

    @Test(expected=IllegalArgumentException.class)
    public void testBadURI() {
        XMLUtil.verifyURI(":junk");
    }

    @Test
    public void testEscape() {
        assertEquals("&lt;", XMLUtil.escape("<"));
        assertEquals("&gt;", XMLUtil.escape(">"));
        assertEquals("&amp;", XMLUtil.escape("&"));
        assertEquals("&apos;", XMLUtil.escape("'"));
        assertEquals("&quot;", XMLUtil.escape("\""));
        assertEquals("1&lt;2&gt;3&amp;4&apos;5&quot;6", XMLUtil.escape("1<2>3&4'5\"6"));
    }

    @Test
    public void testIsComment() {
        assertTrue(XMLUtil.isComment("one two"));
        assertTrue(XMLUtil.isComment("one - two"));
        assertTrue(!XMLUtil.isComment("one -- two"));
        assertTrue(!XMLUtil.isComment("-- one two"));
        assertTrue(!XMLUtil.isComment("one two --"));
    }

    @Test
    public void testIsNMToken() {
        assertTrue(XMLUtil.isNMToken("a1"));
        assertTrue(!XMLUtil.isNMToken("1a"));
    }

    @Test
    public void testIsURI() {
        assertTrue(XMLUtil.isURI("http://www.ociweb.com/foo"));
        assertTrue(!XMLUtil.isURI(":junk"));
    }

    @Test
    public void testIsVersion() {
        assertTrue(XMLUtil.isVersion("1.0"));
        assertTrue(XMLUtil.isVersion("1.1"));
        assertTrue(XMLUtil.isVersion("1.2"));
        assertTrue(!XMLUtil.isVersion("1.3"));
    }
}

File: trunk/c#/WAXTest/XMLUtilTest.cs
using System;
using WAXNamespace;
using Xunit;

namespace WaxTest
{
    public class XMLUtilTest
    {
        [Fact]
        public void testBadComment()
        {
            try {
                XMLUtil.VerifyComment("one -- two");
                Assert.True(false, "Expected ArgumentException.");
            }
            catch (ArgumentException ex)
            {
                Assert.Equal("\"one -- two\" is an invalid comment", ex.Message);
            }
        }

        [Fact]
        public void testBadNMToken()
        {
            try
            {
                XMLUtil.VerifyNMToken("1a");
                Assert.True(false, "Expected ArgumentException.");
            }
            catch (ArgumentException ex)
            {
                Assert.Equal("\"1a\" is an invalid NMTOKEN", ex.Message);
            }
        }

        [Fact]
        public void testBadVersion()
        {
            try
            {
                XMLUtil.VerifyVersion("1.3");
                Assert.True(false, "Expected ArgumentException.");
            }
            catch (ArgumentException ex)
            {
                Assert.Equal("\"1.3\" is an invalid XML version", ex.Message);
            }
        }

        [Fact]
        public void testBadURI()
        {
            try
            {
                XMLUtil.VerifyURI(":junk");
                Assert.True(false, "Expected ArgumentException.");
            }
            catch (ArgumentException ex)
            {
                Assert.Equal("\":junk\" is an invalid URI", ex.Message);
            }
        }

        [Fact]
        public void testEscape() {
            Assert.Equal("&lt;", XMLUtil.Escape("<"));
            Assert.Equal("&gt;", XMLUtil.Escape(">"));
            Assert.Equal("&amp;", XMLUtil.Escape("&"));
            Assert.Equal("&apos;", XMLUtil.Escape("'"));
            Assert.Equal("&quot;", XMLUtil.Escape("\""));
            Assert.Equal("1&lt;2&gt;3&amp;4&apos;5&quot;6", XMLUtil.Escape("1<2>3&4'5\"6"));
        }

        [Fact]
        public void testIsComment() {
            Assert.True(XMLUtil.IsComment("one two"));
            Assert.True(XMLUtil.IsComment("one - two"));
            Assert.True(!XMLUtil.IsComment("one -- two"));
            Assert.True(!XMLUtil.IsComment("-- one two"));
            Assert.True(!XMLUtil.IsComment("one two --"));
        }

        [Fact]
        public void testIsNMToken() {
            Assert.True(XMLUtil.IsNMToken("a1"));
            Assert.True(!XMLUtil.IsNMToken("1a"));
        }

        [Fact]
        public void testIsURI() {
            Assert.True(XMLUtil.IsURI("http://www.ociweb.com/foo"));
            Assert.True(!XMLUtil.IsURI(":junk"));
        }

        [Fact]
        public void testIsVersion() {
            Assert.True(XMLUtil.IsVersion("1.0"));
            Assert.True(XMLUtil.IsVersion("1.1"));
            Assert.True(XMLUtil.IsVersion("1.2"));
            Assert.True(!XMLUtil.IsVersion("1.3"));
        }
    }
}
